packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts

```diff
@@ -128,24 +149,47 @@ export const verifyPresentations = async (
     if (!verifyOpts.verification.revocationOpts?.revocationVerificationCallback) {
       throw Error(`Please provide a revocation callback as revocation checking of credentials and presentations is not disabled`)
     }
-    for (const vp of presentationsArray) {
+    for (const vp of wrappedPresentations) {
       await verifyRevocation(vp, verifyOpts.verification.revocationOpts.revocationVerificationCallback, revocationVerification)
     }
   }
-  return { nonce, presentations: presentationsArray, presentationDefinitions, submissionData: presentationSubmission }
+  if (presentationDefinitions) {
+    return { nonce, presentations: wrappedPresentations, presentationDefinitions, submissionData: presentationSubmission }
+  } else {
+    return { nonce, presentations: wrappedPresentations, dcqlQuery }
+  }
+}
+
+export const extractPresentationRecordFromDcqlVpToken = (
+  vpToken: DcqlPresentationRecord.Input | string,
+  opts?: { hasher?: Hasher },
+): Record<string, WrappedVerifiablePresentation> => {
```

Comment:
I think either making a separate type or using the `{ [credentialQueryId: string]: WrappedVerifiablePresentation }` here can help to understand what the key in the record is.